File: frontend-expo/src/services/followService.js
import api from './api';
import { ENDPOINTS } from '../constants/api';

export const followService = {
    searchUsers: async (userId, query) => {
        const res = await api.get(ENDPOINTS.SEARCH_USERS, { params: { userId, query } });
        return res.data;
    },

    // Bağlantı isteği gönder
    follow: async (userId, targetId) => {
        await api.post(ENDPOINTS.FOLLOW_USER(targetId), {}, { params: { userId } });
    },

    // Bağlantı isteğini geri çek veya bağlantıyı kes
    unfollow: async (userId, targetId) => {
        await api.delete(ENDPOINTS.FOLLOW_USER(targetId), { params: { userId } });
    },

    getFollowing: async (userId) => {
        const res = await api.get(ENDPOINTS.GET_FOLLOWING(userId));
        return res.data;
    },

    getFollowers: async (userId) => {
        const res = await api.get(ENDPOINTS.GET_FOLLOWERS(userId));
        return res.data;
    },

    // Her iki yönden onaylanmış bağlantılar (görev atama için kullanılır)
    getConnections: async (userId) => {
        const res = await api.get(ENDPOINTS.GET_CONNECTIONS(userId));
        return res.data;
    },

    // Gelen onay bekleyen istekler
    getFollowRequests: async (userId) => {
        const res = await api.get(ENDPOINTS.GET_FOLLOW_REQUESTS, { params: { userId } });
        return res.data;
    },

    // Kabul et
    approveFollow: async (followId, userId) => {
        await api.post(ENDPOINTS.APPROVE_FOLLOW(followId), {}, { params: { userId } });
    },

    // Reddet
    rejectFollow: async (followId, userId) => {
        await api.delete(ENDPOINTS.APPROVE_FOLLOW(followId), { params: { userId } });
    },
};

File: backend/src/main/java/com/taskio/service/FollowService.java
package com.taskio.service;

import com.taskio.dto.response.FollowRequestResponse;
import com.taskio.dto.response.UserResponse;
import java.util.List;

public interface FollowService {
    void follow(Long followerId, Long followingId);
    void unfollow(Long followerId, Long followingId);
    void approveFollow(Long followId, Long currentUserId);
    void rejectFollow(Long followId, Long currentUserId);
    List<UserResponse> getFollowing(Long userId);      // Sadece ben takip ettim ACCEPTED
    List<UserResponse> getFollowers(Long userId);      // Beni takip eden ACCEPTED
    List<UserResponse> getConnections(Long userId);    // Her iki yönden bağlantılar (görev atama için)
    List<FollowRequestResponse> getPendingRequests(Long userId);
    List<UserResponse> searchUsers(Long currentUserId, String query);
}

